server/src/main/java/org/opensearch/index/translog/checked/TranslogCheckedContainer.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog.checked;
+
+import org.opensearch.common.io.Channels;
+import org.opensearch.common.util.concurrent.ReleasableLock;
+
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.zip.CRC32;
+import java.util.zip.Checksum;
+
+/**
+ * TranslogCheckedContainer is used to store, update and retrieve checksums for translog files.
+ *
+ * @opensearch.internal
+ */
+public class TranslogCheckedContainer {
```

Comment:
Instead of creating new class to calculate checksum , can we add a function in `FileSnapshot` to return `Checksum` ?

Reply:
As per the new code which updates translog checksum during each operation append instead of calculating the checksum once for the entire file during upload, we need to maintain a construct decoupled from `FileSnapshot` to store an in memory checksum for the latest translog and checkpoint file.